engine/api/src/main/scala/pl/touk/nussknacker/engine/api/definition/ParameterValidator.scala

```diff
@@ -98,6 +98,56 @@ case object LiteralIntegerValidator extends ParameterValidator {
   )
 }
 
+case class MinimalNumberValidator(minimalNumber: BigDecimal, annotationMessage: String) extends ParameterValidator {
+
+  private lazy val minAnnotationDefaultMessage: String = "{javax.validation.constraints.Min.message}"
+  private lazy val minAnnotationDefaultMessageRegex: String = s"^${minAnnotationDefaultMessage}" + "$"
+  private lazy val defaultValidatorMessage: String = s"This field value has to be a number greater than or equal to $minimalNumber"
+
+  //Blank value should be not validate - we want to chain validators
+  override def isValid(paramName: String, value: String, label: Option[String])(implicit nodeId: NodeId): Validated[PartSubGraphCompilationError, Unit] =
+    if (StringUtils.isBlank(value) || Try(BigDecimal(value)).filter(_ >= minimalNumber).isSuccess)
+      valid(Unit)
+    else
+      invalid(error(paramName, nodeId.id))
+
+  private def error(paramName: String, nodeId: String): InvalidNumberParameter = InvalidNumberParameter(
+    getErrorMessageOrDefault(),
+    "Please fill field by proper number",
+    paramName,
+    nodeId
+  )
+
+  private def getErrorMessageOrDefault() = {
+    annotationMessage.replace(minAnnotationDefaultMessageRegex, defaultValidatorMessage)
+  }
+}
+
+case class MaximalNumberValidator(maximalNumber: BigDecimal, annotationMessage: String) extends ParameterValidator {
+
+  private lazy val maxAnnotationDefaultMessage: String = "{javax.validation.constraints.Max.message}"
+  private lazy val maxAnnotationDefaultMessageRegex: String = s"^${maxAnnotationDefaultMessage}" + "$"
+  private lazy val defaultValidatorMessage: String = s"This field value has to be a number less than or equal to $maximalNumber"
+
+  //Blank value should be not validate - we want to chain validators
+  override def isValid(paramName: String, value: String, label: Option[String])(implicit nodeId: NodeId): Validated[PartSubGraphCompilationError, Unit] =
+    if (StringUtils.isBlank(value) || Try(BigDecimal(value)).filter(_ <= maximalNumber).isSuccess)
```

Comment:
I think that we can extract some `BaseNumberValidator` and (or at least) `AcceptingOptionalValueValidator`? To avoid this `if isBlank || part`

Reply:
We should have two strategies:
- For required and non empty 
- Else validators
We should create another issue for it..